### ISSUE TITLE
The Pipeline format is too cumbersome

### ISSUE THREAD

**Issue by BoyChai** [tektoncd/pipeline#7977]
### Feature request
In the process of writing Pipeline, it is hoped that it can be directly written through workspaces and params.
Although this is not as good as writing through pipelinerun and TriggerTemplate, I hope to support this feature by directly passing values in the params and workspaces in Pipeline. Because sometimes small-scale projects don't seem to require such deep decoupling, and multiple layers can appear cumbersome and complex.
I don't want to use this feature, but I hope it can't be without it, because writing configurations like this can make it more convenient for some small-scale applications.

### Use case
```pipline.yaml
---
apiVersion: tekton.dev/v1
kind: Pipeline
metadata:
  name: xxx-pipeline
  namespace: cicd-services
spec:
  workspaces:
    - name: remote-ssh-secret
      secret:
        secretName: remote-ssh-secret
    - name: remote-scp-secret
      secret:
        secretName: remote-scp-secret
    - name: go-cache
      persistentVolumeClaim:
        claimName: go-cache-pvc
    - name: repo-pvc
      persistentVolumeClaim:
        claimName: xxx-source-pvc
  params:
    - name: url
      type: string
      value: "xxx.git"
    - name: branch
      type: string
      value: "master"
    - name: backDir
      type: string
      value: "backend"
    - name: goImgTag
      type: string
      value: "1.19.0"
    - name: remoteHost
      type: string
      value: "xxx.xxx.xxx"
    - name: remoteUser
      type: string
      value: "root"
    - name: remotePort
      type: string
      value: "22"
  tasks:
    ...
```

**Comment by chitrangpatel** [comment]
@BoyChai Do you think this helps? https://github.com/tektoncd/pipeline/issues/7901 It was released in v0.60.0

**Comment by BoyChai** [comment]
> @BoyChai Do you think this helps? #7901 It was released in v0.60.0

I think it's very useful because some small-scale tasks don't require so many layers of Params and Workspaces, and such multi-layered tasks can become more complex. Thank you for your PR, I will close the issue now
